### PR TITLE
Package ProGuard rules into the aar.

### PR DESCRIPTION
**Pull request by monxalo**

### CODE REVIEW

**Comment by andhie** [comment]
This prevents the whole library from being Proguard.
Should use a stricter rule. Solution specified in https://github.com/PhilJay/MPAndroidChart/issues/348#issuecomment-87453245

```java
-keep public class com.github.mikephil.charting.animation.* {
    public protected *;
}
```

**Comment by monxalo** [comment]
I just followed the proguard rules from the [wiki](https://github.com/PhilJay/MPAndroidChart/wiki/Proguard), which says that it needs to whitelist the library but maybe there are better rules.

**Comment by hannesa2** [comment]
@monxalo and @andhie Thank you, for me it works.
Because the maintainer is not very fast with the merge, I helped myself and if other want this solution too, he has to 

```
allprojects {
	repositories {
		...
		maven { url 'https://jitpack.io' }
	}
}
```

and 

` implementation 'com.github.hannesa2:MPAndroidChart:3.0.5'`

**Comment by andhie** [comment]
you need to remove the original rule `-keep class com.github.mikephil.charting.** { *; }`
if you have this, it still means keeping everything even you have a stricter rule

**Comment by hannesa2** [comment]
Thank you for the hint
https://github.com/hannesa2/MPAndroidChart/commit/115cdb9105e6e496976c0b778cb4bfaa469ba87f

**Comment by almic** [comment]
This doesn't seem necessary with Android Studio, is this specific to Eclipse or some other thing? I deleted these files in a recent commit because it was causing the Android lint to give warnings.

**Comment by sonique6784** [comment]
@monxalo can you please fix conflict?

File: MPChartLib/proguard-project.txt [hunk 1]
```diff
@@ -1,20 +1,4 @@
...
 
-# Add any project specific keep options here:
-
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
+-keep class com.github.mikephil.charting.** { *; }
```
Comment (sonique6784) [inline MPChartLib/proguard-project.txt]: following rules should be used:
`-dontwarn com.github.mikephil.**`

`-keep public class com.github.mikephil.charting.animation.* {
    public protected *;
}`

**Pull request closed without merging**